core/circuit.go

```diff
@@ -0,0 +1,262 @@
+package core
+
+import (
+	"fmt"
+	"math"
+	"strings"
+	"time"
+
+	"github.com/evcc-io/evcc/api"
+	"github.com/evcc-io/evcc/util"
+)
+
+var gridMeterUsed bool // indicates gridmeter is used already fo a circuit to avoid > 1 usage
+
+type Circuit struct {
+	Log    *util.Logger
+	uiChan chan<- util.Param
+
+	Name       string    `mapstructure:"name"`       // meaningful name
+	MaxCurrent float64   `mapstructure:"maxCurrent"` // the max allowed current of this circuit
+	MeterRef   string    `mapstructure:"meter"`      // Charge meter reference
+	Circuits   []Circuit `mapstructure:"circuits"`   // sub circuits as config reference
+
+	parentCircuit *Circuit         // parent circuit reference
+	meterCurrent  api.MeterCurrent // meter to determine phase current
+	vMeter        *VMeter          // virtual meter if no real meter is used
+}
+
+// determines current in use.
+// implement consumer interface
+// TBD: phase perfect evaluation
+func (cc *Circuit) GetCurrent() (float64, error) {
+	if cc.meterCurrent == nil {
+		cc.Log.ERROR.Println("meterCurrent is nil")
+		return 0.0, fmt.Errorf("no meter available")
+	}
+
+	var current float64
+	var err error
```

Comment:
DU musst keine Variablen deklarieren die hinterher in einem assignment erzeugt werden